src/Server/modules/WorldPlaceManager.js

```diff
@@ -0,0 +1,401 @@
+const dataClassesModule = require('../modules/DataClasses.js');
+const readCSVModule = require('../modules/ReadCSV.js');
+const writeCSVModule = require('../modules/WriteCSV.js');
+const helper = require('../modules/BasicHelpers.js');
+const analytics = require('../modules/Analytics.js');
+const worldReq = require('../modules/DataClasses.js');
+
+// Array of WorldPlaces read from CSV
+var allWorldPlace = [];
+
+// Populates allWorldPlace with data read from CSV
+function InitializeAllWorldPlace() {
+	allWorldPlace = readCSVModule.LoadWorldData();
+}
+
+function AddWorldData(province, country, date, tod, number) {
+	if (allWorldPlace.length == 0) InitializeAllWorldPlace();
+
+    var errormsg = 'no error';
+	if (tod == 1) {
+		for (var i = 0; i < allWorldPlace.length; ++i) {
+			if (country == allWorldPlace[i].country && province == allWorldPlace[i].state) {
+				for (var key in allWorldPlace[i].numConfirmed) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allWorldPlace[i].numConfirmed[temp_date] = number;
+				allWorldPlace[i].currentNumConfirmed = number;
```

Comment:
We want to change currentNumConfirmed only when temp_date is the most recent date of all existing dates.